src/ansys/dpf/core/mesh_info.py

```diff
@@ -0,0 +1,166 @@
+            elif gdc is not None and mesh_info is None:
+                self._gdc = gdc
+            elif gdc is None and MeshInfo is not None:
+                self._gdc = mesh_info._gdc
+        except ValueError:
+            print("Both generic data container and mesh info can't be filled")
+
+    def __call__(self):
+        return self
+
+    def deep_copy(self, server=None):
+        """Create a deep copy of the scoping's data on a given server.
+
+        This method is useful for passiong data from one server instance to another.
+
+        Parameters
+        ----------
+        server : ansys.dpf.core.server, optional
+            Server with the channel connected to the remote or local instance.
+            The default is ``None``, in which case an attempt is made to use the
+            global server.
+
+        Returns
+        -------
+        scoping_copy : Scoping
+        """
+        mesh_info = MeshInfo(server=server)
+        mesh_info._gdc = self._gdc
+        return mesh_info
+
+    def get_property(self, property_name, output_type):
+        """Get property with given name.
+
+        Parameters
+        ----------
+        property_name : str
+            Property name.
+        output_type :  :class:`ansys.dpf.core.common.types`
+
+        Returns
+        -------
+        type
+            Property object instance.
+        """
+
+        return self._gdc.get_property(property_name, output_type)
+
+    def set_property(self, property_name, prop):
+        """Register given property with the given name.
+
+        Parameters
+        ----------
+        property_name : str
+            Property name.
+        prop :  type of the property.
+
+        Returns
+        -------
+        type
+            Property object instance.
+        """
+
+        return self._gdc.set_property(property_name, prop)
+
+    def get_number_nodes(self):
+        """
+        Returns
+        -------
+        number_nodes : int
+            Number of nodes of the mesh.
+        """
+
+        return self._gdc.get_property("num_nodes", int)
+
+    def get_number_elements(self):
+        """
+        Returns
+        -------
+        number_elements : int
+            Number of elements of the mesh.
+        """
+
+        return self._gdc.get_property("num_elements", int)
+
+    def get_splittable_by(self):
+        """
+        Returns
+        -------
+        splittable by which entity : StringField
+            Number of elements of the mesh.
+        """
+
+        return self._gdc.get_property("splittable_by", ansys.dpf.core.StringField)
+
+    def get_available_elem_types(self):
+        """
+        Returns
+        -------
+        available element types : Scoping
+            element type available for the mesh.
+        """
+
+        return self._gdc.get_property("avalaible_elem_type", ansys.dpf.core.Scoping)
+
+    def set_number_nodes(self, number_of_nodes):
+        """Number of nodes"""
```

Comment:
The docstrings of the setters should reflect what they bare setting. For example: https://github.com/ansys/pydpf-core/blob/709032569e01dec991c79ae6ff25c11a862fdbec/src/ansys/dpf/core/collection.py#L118C9-L119C1